src/FileLocator.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+namespace RebelCode\Modular\Locator;
+
+use Dhii\Modular\Locator\ModuleLocatorInterface;
+use Dhii\Modular\Locator\CouldNotReadSourceExceptionInterface;
+use Dhii\Validation\ValidatorInterface;
+use Dhii\Validation\Exception\ValidationFailedExceptionInterface;
+use Dhii\I18n\StringTranslatorAwareTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use RebelCode\Modular\Config\ConfigInterface as Cfg;
+use Traversable;
+
+/**
+ * A module config locator which reads configuration from files.
+ *
+ * @since [*next-version*]
+ */
+class FileLocator extends AbstractFileLocator implements ModuleLocatorInterface
+{
+    /**
+     * The default maximal depth of JSON files to read.
+     *
+     * @since [*next-version*]
+     */
+    const JSON_DEFAULT_MAX_DEPTH = 512;
+
+    /*
+     * @since [*next-version*]
+     */
+    use StringTranslatorAwareTrait;
+
+    /*
+     * @since [*next-version*]
+     */
+    use StringTranslatingTrait;
+
+    /**
+     * The validator used to validate configuration.
+     *
+     * @since [*next-version*]
+     *
+     * @var ValidatorInterface
+     */
+    protected $configValidator;
+
+    /**
+     * @since [*next-version*]
+     *
+     * @param array|Traversable  $sources   The list of config sources, each of which is a file path.
+     * @param ValidatorInterface $validator The validator to use for config validation.
+     */
+    public function __construct($sources, ValidatorInterface $validator)
+    {
+        $this->_setSources($sources);
+        $this->_setConfigValidator($validator);
+        $this->_construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _read($source)
+    {
+        $ext = $this->_getFileExtension($source);
+        if ($ext === 'php') {
```

Comment:
I thought about it, and wanted to map extensions. I thought again, and wanted to map file types. Then I realized that this is the best way for now, because it's simple, and would allow us to easily cater for some un-mappable case. When there is sufficient usage data to see how to improve it, then we improve.